src/main/java/homework9/LinkedListImplementation.java

```diff
@@ -0,0 +1,209 @@
+package homework9;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LinkedListImplementation <T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public LinkedListImplementation() {
+        this.size = 0;
+    }
+
+    private static class Node<T> {
+        Node<T> previous;
+        T item;
+        Node<T> next;
+
+        public Node(Node<T> previous, T item, Node<T> next) {
+            this.previous = previous;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    public void linkFirst(T node) {
```

Comment:
better make this method private and rename  parameter to value

we can simplify this method:
Node<T> newNode = new Node<>(null, value, null);
first = newNode;
last = first;
size++;